src/MentionsInput.js

```diff
@@ -311,16 +311,17 @@ class MentionsInput extends React.Component {
     let newPlainTextValue = ev.target.value
 
     // Derive the new value to set by applying the local change in the textarea's plain text
-    let newValue = applyChangeToValue(
-      value,
-      markup,
-      newPlainTextValue,
-      this.state.selectionStart,
-      this.state.selectionEnd,
-      ev.target.selectionEnd,
-      displayTransform,
-      regex
-    )
+    let newValue = newPlainTextValue === '@' ?
```

Comment:
This assumes that mentions are only triggered using `@` which is not the case. If I understand the approach correctly then this should take all possible trigger characters into account.

Reply:
Also `newPlainTextValue` is the entire value in the textarea. So it would only fix the problem in the case the textarea is completely empty. I'm afraid this PR doesn't really bring us any further in fixing the problem.